tests/components/datadog/test_init.py

```diff
@@ -51,20 +56,31 @@ async def test_datadog_setup_defaults(hass: HomeAssistant) -> None:
             },
         )
 
-        assert mock_init.call_count == 1
-        assert mock_init.call_args == mock.call(statsd_host="host", statsd_port=8125)
+        assert mock_dogstatsd.call_count == 1
+        assert mock_dogstatsd.call_args == mock.call("host", 8125)
 
 
 async def test_logbook_entry(hass: HomeAssistant) -> None:
     """Test event listener."""
     with (
-        patch("homeassistant.components.datadog.initialize"),
-        patch("homeassistant.components.datadog.statsd") as mock_statsd,
+        patch("homeassistant.components.datadog.DogStatsd") as mock_statsd_class,
+        patch(
+            "homeassistant.components.datadog.config_flow.DogStatsd", mock_statsd_class
+        ),
     ):
+        mock_statsd = mock_statsd_class.return_value
+
         assert await async_setup_component(
             hass,
             datadog.DOMAIN,
-            {datadog.DOMAIN: {"host": "host", "rate": datadog.DEFAULT_RATE}},
+            {
+                datadog.DOMAIN: {
```

Comment:
In a a follow-up PR it would be good to change all these tests to load from config entry instead so a clean-up of the yaml config is easier when it's time for that.

Reply:
Yeah, rather easy to just make a config entry and load that instead of the yaml config.
but let's do that in a follow-up